xbmc/addons/Addon.h

```diff
@@ -56,38 +56,38 @@ class CAddon : public IAddon
   explicit CAddon(CAddonInfo addonInfo);
   virtual ~CAddon() {}
 
-  TYPE Type() const override { return m_addonInfo.type; }
+  TYPE Type() const override { return m_addonInfo.MainType(); }
```

Comment:
In the end though there really should be no such thing as an 'addon type', but only addons with N possible extension points (which can have a type). So assuming this is only temporary for legacy code I guess 'MainType' works as to making it clearer..